kitchen/bin/kitchen

```diff
@@ -0,0 +1,901 @@
+            if termination_delay_ms >= 0:
+                source = 'header after {} milliseconds'.format(termination_delay_ms)
+                run_after_ms(termination_delay_ms, terminate, source)
+
+        # Echo request body back to client
+        if self.headers.get('x-kitchen-echo') is not None:
+            post_bytes_count = int(self.headers.get('content-length', 0))
+            self.__set_response(self.rfile.read(post_bytes_count))
+
+        # Slurp any posted data (non-echo mode only)
+        else:
+            content_length = self.headers.get('content-length')
+            if content_length is not None:
+                self.__slurp_bytes(int(content_length))
+
+            elif self.headers.get('transfer-encoding') == 'chunked':
+                while True:
+                    chunk_header = self.rfile.read(3)  # shortest possible header is b'0\r\n'
+
+                    while not chunk_header.endswith(b'\r\n'):
+                        chunk_header += self.rfile.read(1)
+
+                    chunk_size = int(chunk_header[:-2], base=16)
+                    self.__slurp_bytes(chunk_size + 2)  # read the payload + '\r\n'
+
+                    if chunk_size == 0:
+                        break
+
+        # Exclude some headers from response
+        excluded_headers = self.headers.get('x-kitchen-exclude-headers')
+        if excluded_headers:
+            self.__excluded_headers = set(x.lower() for x in excluded_headers.split(','))
+
+        # Response payload length
+        truncated_size = self.headers.get('x-kitchen-fail-after')
+        if truncated_size:
+            self.__truncated_length = int(truncated_size)
+
+        # Kill the whole server on failure
+        if self.headers.get('x-kitchen-fail-by-terminating-jvm') is not None:
+            self.__exit_process = True
+
+        # Asynchronous resource request id
+        req_id = self.headers.get('x-kitchen-request-id')
+        if req_id and self.__async_req:
+            self.__async_req['id'] = req_id
+
+        # Response payload length
+        response_size = self.headers.get('x-kitchen-response-size')
+        if response_size:
+            self.__response_length = int(response_size)
+
+        # Response payload length
+        async_expire_ms = self.headers.get('x-kitchen-store-async-response-ms')
+        if async_expire_ms and self.__async_req:
+            self.__async_req['expire-ms'] = int(async_expire_ms)
+
+    def __process_path(self):
+        """Handle logic for all supported Kitchen endpoint paths."""
+        path = self.__path
+        query_string = self.__query
+
+        if query_string:
+            query_params = dict(split2(param, '=', 1) for param in query_string.split('&'))
+        else:
+            query_params = {}
+
+        if path == '/async/request':
+            self.__async_req = {
+                    'delay-ms': 20000,
+                    'expire-ms': 10000,
+                    'type': 'create'
+                    }
```

Comment:
Location of the closing brace seems odd

Reply:
I tried this several different ways, and I think this looks the most natural (at least with my default vim indenting). Apparently PEP8 doesn't address this. However, I did just discover that Python supports a trailing comma on the last entry, so I added a trailing comma to each of these multiline dictionary literals. Having the open-brace, each entry, and the close-brace all on separate lines makes it easier to insert/remove/resort the entries later without affecting as many lines with the diff.